src/Container/Container.php

```diff
@@ -36,76 +41,128 @@ class Container implements ContainerInterface
      * @param mixed  $value
      * @return Container
      */
-    public function set($key, $value)
+    public function set(string $key, $value): self
     {
+        if (!(\is_string($value) || \is_object($value))) {
+            throw new \InvalidArgumentException('Value must be a class name, an object instance, or a callable');
+        }
+
         $this->dicValues[$key] = $value;
         unset($this->dicObjects[$key]); // In case an object instance was stored for sharing
 
         return $this;
     }
 
+    public function alias(string $alias, string $key): self {
+        $this->aliases[$alias] = $key;
+
+        return $this;
+    }
+
     /**
      * Check if a DIC value/object exists.
      *
      * @param string $key
      * @return bool
      */
-    public function has($key)
+    public function has($key): bool
```

Comment:
parameter type hint missing: `     public function has(string $key): bool`

Reply:
Unfortunately this is not compatible with Psr\Container\ContainerInterface

Reply:
argh ... yes of course